clients/src/main/java/org/apache/kafka/common/requests/AbstractRequest.java

```diff
@@ -76,54 +88,78 @@ public short version() {
     /**
      * Factory method for getting a request object based on ApiKey ID and a buffer
      */
-    public static AbstractRequest getRequest(int requestId, short versionId, ByteBuffer buffer) {
+    public static RequestAndSize getRequest(int requestId, short version, ByteBuffer buffer) {
```

Comment:
I guess another option is to add the request size as a field in `ProduceRequest`, since that appears to be the only usage.